packages/execution-environments/src/common/keyring.ts

```diff
@@ -0,0 +1,24 @@
+import { SnapKeyring } from '@metamask/snap-types';
+
+/**
+ * Wraps a SnapKeyring class, returning a handler that can route requests to the exposed functions by the class.
+ *
+ * @param keyring - The SnapKeyring instance.
+ * @returns A handler function.
+ */
+export function wrapKeyring(keyring?: SnapKeyring) {
+  if (!keyring) {
+    throw new Error('Keyring not exported');
+  }
+
+  const keyringHandler = ({ request }: any) => {
+    const { method, params } = request;
+    if (!(method in keyring)) {
```

Comment:
Use `hasProperty`.

Reply:
```
Argument of type 'SnapKeyring' is not assignable to parameter of type 'RuntimeObject'
```
🤔

Reply:
`in` checks the prototype which what we want to do in this case, no?

Reply:
MetaMask/eslint-config#207

Reply:
`in` is what we want for this specific case.